pkg/operator/saptuneapplysolution_v1.go

```diff
@@ -81,123 +102,64 @@ func NewSaptuneApplySolution(
 		return nil
 	}
 
-	applyOutput, err := sa.executor.Exec(ctx, "saptune", "solution", "apply", sa.parsedArguments.solution)
-	if err != nil {
-		sa.logger.Errorf(
-			"could not perform saptune solution apply %s, error output: %s",
-			sa.parsedArguments.solution,
-			applyOutput,
-		)
-
-		return fmt.Errorf("could not perform the saptune apply solution %s, error: %s",
+	if initiallyAppliedSolution != "" {
+		return fmt.Errorf(
+			"cannot apply solution %s because another solution %s is already applied",
 			sa.parsedArguments.solution,
-			err,
+			initiallyAppliedSolution,
 		)
 	}
 
-	return nil
+	return sa.saptune.ApplySolution(ctx, sa.parsedArguments.solution)
 }
 
 func (sa *SaptuneApplySolution) verify(ctx context.Context) error {
-	solutionAppliedOutput, err := sa.executor.Exec(ctx, "saptune", "--format", "json", "solution", "applied")
+	appliedSolution, err := sa.saptune.GetAppliedSolution(ctx)
 	if err != nil {
-		return errors.New("could not call saptune solution applied")
-	}
-
-	if alreadyApplied := isSaptuneSolutionAlreadyApplied(solutionAppliedOutput, sa.parsedArguments.solution); alreadyApplied {
-		sa.resources[afterDiffField] = string(solutionAppliedOutput)
-		return nil
+		return err
 	}
 
-	return fmt.Errorf(
-		"verify saptune apply failing, the solution %s was not applied in commit phase",
-		sa.parsedArguments.solution,
-	)
-}
-
-func (sa *SaptuneApplySolution) rollback(ctx context.Context) error {
-	revertOutput, err := sa.executor.Exec(ctx, "saptune", "solution", "revert", sa.parsedArguments.solution)
-	if err != nil {
-		return fmt.Errorf("could not revert saptune solution %s during rollback, error: %s",
+	if appliedSolution != sa.parsedArguments.solution {
+		return fmt.Errorf(
+			"verify saptune apply failing, the solution %s was not applied in commit phase",
 			sa.parsedArguments.solution,
-			revertOutput,
 		)
 	}
-
+	sa.resources[afterDiffField] = appliedSolution
 	return nil
 }
 
-func isSaptuneVersionSupported(version string) bool {
-	compareOutput := semver.Compare(minimalSaptuneVersion, "v"+version)
-
-	return compareOutput != 1
-}
-
-func isSaptuneSolutionAlreadyApplied(saptuneOutput []byte, solution string) bool {
-	return gjson.GetBytes(saptuneOutput, fmt.Sprintf(`result.Solution applied.#(Solution ID=="%s")`, solution)).Exists()
-}
-
-func parseSaptuneApplyArguments(rawArguments OperatorArguments) (*saptuneApplySolutionArguments, error) {
-	argument, found := rawArguments["solution"]
-	if !found {
-		return nil, errors.New("argument solution not provided, could not use the operator")
-	}
-
-	solution, ok := argument.(string)
-	if !ok {
-		return nil, fmt.Errorf(
-			"could not parse solution argument as string, argument provided: %v",
-			argument,
-		)
-	}
-
-	return &saptuneApplySolutionArguments{
-		solution: solution,
-	}, nil
+func (sa *SaptuneApplySolution) rollback(ctx context.Context) error {
+	return sa.saptune.RevertSolution(ctx, sa.parsedArguments.solution)
```

Comment:
I think here we should better not run `revert` if we already have other solution installed.
I know that the command wouldn't do any harm, as we are trying to revert other solution, but sparing any command in saptune is good.

Reply:
I wouldn't mind adding an if, however isn't that different from what Angela said? 🤔
I understood that: if applying a solution fails the best option would be revering it.

(And yes we'd be reverting the requested one)

Reply:
But we don't even try to apply the solution, we fail before, as there is already other solution applied. `cannot apply solution %s because another solution %s is already applied` code branch.
In this case, we don't need to revert anything, right?

Reply:
Of course 😄 Thanks! 7d0e6ca1b9029361a45bfab010e71aea0f1508dd